apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -1695,12 +1696,14 @@ protected function canDeleteShareFromSelf(\OCP\Share\IShare $share): bool {
 	private function parseDate(string $expireDate): \DateTime {
 		try {
 			$date = new \DateTime(trim($expireDate, "\""), $this->dateTimeZone->getTimeZone());
+			// Make sure it expires at midnight in user timezone
```

Comment:
Maybe less confusing for our future selves.

```suggestion
			// Make sure it expires at midnight in owner timezone
```

Reply:
Updated!